.travis.yml

```diff
@@ -43,7 +27,46 @@ jobs:
         # Travis. This can be slow to install, so raise the Travis timeout.
         - (which rustfmt && rustfmt --version && [[ "$(rustfmt --version)" =~ "0.2.9" ]]) || travis_wait cargo install --force rustfmt-nightly --vers 0.2.9
       script: ./.travis-format.sh
+    - &FULL_TEST
+      stage: test
+      env:
+        # Ensure that we build without warnings.
+        - CARGO_FLAGS="--features 'strict'"
       os: linux
+      before_script:
+        - sudo apt install -y texinfo libgif-dev libxpm-dev
+      script:
+        - ./autogen.sh && ./configure
+        - make -j 3 && echo '==> Running tests' && make check
+    -
+      <<: *FULL_TEST
+      os: osx
+      before_script:
+        - brew install gnutls
+        - brew install texinfo
+    - &NO_WINDOW_SYSTEM_TEST
+      stage: test
+      env:
+        # Ensure that we build without warnings.
+        - CARGO_FLAGS="--features 'strict'"
+        - TRAVIS_WITH_NO_WINDOW_SYSTEM=1  # So we can tell them apart
+        - TRAVIS_ALLOW_FAILURE=1
+      os: linux
+      addons:
+        apt:
+          packages:
+            - texinfo
+      before_script:
+        - if [ $TRAVIS_OS_NAME = linux ]; then ./autogen.sh && ./configure --without-makeinfo --with-x=no --without-gconf --without-gsettings; fi
+        # TODO: add the necessary dependencies so we can use the same configure flags on OS X.
+        - if [ $TRAVIS_OS_NAME = osx ]; then ./autogen.sh && ./configure --without-makeinfo --with-gnutls=no --with-x=no --without-gconf --without-gsettings; fi
```

Comment:
gnutls could be installed in this run, too? then the configure options would be the same.

Reply:
I need to validate that brew actually worked. I have been busy on my other PR :-)

Reply:
/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! You're running 2.0.0. (RuntimeError)

And it did not succeed. But it should be possible.